source/diet/html.d

```diff
@@ -290,16 +290,22 @@ private string getElementMixin(ref CTX ctx, in Node node)
 				continue;
 			}
 
-			ret ~= ctx.statement(node.loc, q{static if (is(typeof(%s) == bool)) }~'{', expr);
-			if (ctx.isHTML5) ret ~= ctx.statement(node.loc, q{if (%s) %s.put(" %s");}, expr, ctx.rangeName, att.name);
-			else ret ~= ctx.statement(node.loc, q{if (%s) %s.put(" %s=\"%s\"");}, expr, ctx.rangeName, att.name, att.name);
-			ret ~= ctx.statement(node.loc, "} else "~q{static if (is(typeof(%s) : const(char)[])) }~"{{", expr);
-			ret ~= ctx.statement(node.loc, q{  auto val = %s;}, expr);
-			ret ~= ctx.statement(node.loc, q{  if (val !is null) }~'{');
-			ret ~= ctx.rawText(node.loc, " "~att.name~"=\"");
-			ret ~= ctx.statement(node.loc, q{    %s.filterHTMLAttribEscape(val);}, ctx.rangeName);
-			ret ~= ctx.rawText(node.loc, "\"");
-			ret ~= ctx.statement(node.loc, "  }");
+			ret ~= ctx.statement(node.loc, q{
+				import std.traits : ReturnType;
+				static if (is(typeof(%s) == bool) || is(ReturnType!(typeof(%s)) == bool))
+			}~'{', expr, expr);
```

Comment:
I could have used `%$3s` here as well, but imho this creates a huge dependency on the actual implementation of `statement`.